### PR TITLE
Graph page url client side migration

### PR DESCRIPTION
**Pull request by grandbora**
This pr adds backward compatibility for the changes I introduced at https://github.com/prometheus/prometheus/pull/1872.

The changes lacks any unit tests, this projects has no setup for javascript unit tests. In order to better test this I tried to implement the redirection on the backend (see https://github.com/grandbora/prometheus/commit/802a0a61727e9c8ebad13842b776865fd42ba443) however fragments are not sent to the server :( That's my excuse.

I did manual tests via copying some grafana links and running them on my local instance. I couldn't see any problems, page was redirected to the corresponding new url format.

Please 👀  @juliusv @fabxc @grobie @beorn7

### CODE REVIEW

File: web/ui/static/js/graph.js [hunk 1]
```diff
@@ -777,10 +777,48 @@ function init() {
     url: PATH_PREFIX + "/static/js/graph_template.handlebar",
     success: function(data) {
       graphTemplate = Handlebars.compile(data);
-      var Page = new Prometheus.Page();
-      Page.init();
+      if (isDeprecatedGraphUrl()) {
```
Comment (juliusv) [inline web/ui/static/js/graph.js]: Url -> URL :P (SCNR)

File: web/ui/static/js/graph.js [hunk 2]
```diff
@@ -777,10 +777,48 @@ function init() {
...
   });
 }
 
+
+/**
+ * These two methods (isDeprecatedGraphUrl and redirectToMigratedUrl)
+ * are added only for backward compatibility to old query format.
+ */
+function isDeprecatedGraphUrl() {
+  if (window.location.hash.length = 0) {
```
Comment (juliusv) [inline web/ui/static/js/graph.js]: Is that supposed to be a `=`? Shouldn't that be a `==` or `===`?

File: web/ui/static/js/graph.js [hunk 3]
```diff
@@ -777,10 +777,48 @@ function init() {
...
+      }
     }
   });
 }
 
+
+/**
+ * These two methods (isDeprecatedGraphUrl and redirectToMigratedUrl)
+ * are added only for backward compatibility to old query format.
+ */
```
Comment (juliusv) [inline web/ui/static/js/graph.js]: We don't do block comments anywhere else, so I would stick to `//`.

File: web/ui/static/js/graph.js (hunk 2 above)
Comment (grandbora) [inline web/ui/static/js/graph.js]: Fixed this, together with other remarks.

**Comment by juliusv** [comment]
Woot, thanks!

**Comment by grandbora** [comment]
Thank you!

File: web/ui/static/js/graph.js [hunk 4]
```diff
@@ -777,10 +777,47 @@ function init() {
...
+  try {
+      JSON.parse(decodedFragment.substr(1)); // drop the hash #
+  } catch (e) {
+      return false;
+  }
+  return true;
+}
+
+function redirectToMigratedURL() {
+  var decodedFragment = decodeURIComponent(window.location.hash);
```
Comment (aecolley) [inline web/ui/static/js/graph.js]: The `.substr(1)` is missing here, but it was considered important on line 801.

**Pull request merged**